src/Kestrel.Core/Internal/Http2/HPack/HPackDecoder.cs

```diff
@@ -197,8 +211,14 @@ public void OnByte(byte b, IHttpHeadersHandler handler)
                     {
```

Comment:
I think the new error check should go here rather than in the Decode method. headersObserved needs to become a member field that's reset at the end of Decode if endHeaders is set. That way headersObserved persists across calls to Decode (e.g. Continuation frames).

Reply:
I see.